x/mm2/keeper/keeper_test.go

```diff
@@ -0,0 +1,182 @@
+		types.StoreKey: storetypes.NewTransientStoreKey("transient_mm"),
+	}
+
+	s.authority = sdk.AccAddress("authority")
+	s.ctx = testutil.DefaultContextWithKeys(keys, transientKeys, nil).WithBlockHeight(10)
+
+	k := keeper.NewKeeper(mmSS, encCfg.Codec, s.authority)
+	s.Require().NoError(k.SetLastUpdated(s.ctx, uint64(s.ctx.BlockHeight())))
+
+	params := types.Params{
+		MarketAuthorities: []string{s.authority.String()},
+	}
+	s.Require().NoError(k.SetParams(s.ctx, params))
+
+	return k
+}
+
+func (s *KeeperTestSuite) SetupTest() {
+	s.keeper = s.initKeeper()
+}
+
+func TestKeeperTestSuite(t *testing.T) {
+	suite.Run(t, new(KeeperTestSuite))
+}
+
+var (
+	btcusdt = types.Market{
+		Ticker: types.Ticker{
+			CurrencyPair: slinkytypes.CurrencyPair{
+				Base:  "BITCOIN",
+				Quote: "USDT",
+			},
+			Decimals:         8,
+			MinProviderCount: 1,
+		},
+		ProviderConfigs: []types.ProviderConfig{
+			{
+				Name:           "kucoin",
+				OffChainTicker: "btc-usdt",
+			},
+		},
+	}
+
+	usdtusd = types.Market{
+		Ticker: types.Ticker{
+			CurrencyPair: slinkytypes.CurrencyPair{
+				Base:  "USDT",
+				Quote: "USD",
+			},
+			Decimals:         8,
+			MinProviderCount: 1,
+		},
+		ProviderConfigs: []types.ProviderConfig{
+			{
+				Name:           "kucoin",
+				OffChainTicker: "usdt-usd",
+			},
+		},
+	}
+
+	usdcusd = types.Market{
+		Ticker: types.Ticker{
+			CurrencyPair: slinkytypes.CurrencyPair{
+				Base:  "USDC",
+				Quote: "USD",
+			},
+			Decimals:         8,
+			MinProviderCount: 1,
+		},
+		ProviderConfigs: []types.ProviderConfig{
+			{
+				Name:           "kucoin",
+				OffChainTicker: "usdc-usd",
+			},
+		},
+	}
+
+	ethusdt = types.Market{
+		Ticker: types.Ticker{
+			CurrencyPair: slinkytypes.CurrencyPair{
+				Base:  "ETHEREUM",
+				Quote: "USDT",
+			},
+			Decimals:         8,
+			MinProviderCount: 1,
+		},
+		ProviderConfigs: []types.ProviderConfig{
+			{
+				Name:           "kucoin",
+				OffChainTicker: "eth-usdt",
+			},
+		},
+	}
+
+	markets = []types.Market{
+		btcusdt,
+		usdcusd,
+		usdtusd,
+		ethusdt,
+	}
+
+	marketsMap = map[string]types.Market{
+		btcusdt.Ticker.String(): btcusdt,
+		usdcusd.Ticker.String(): usdcusd,
+		usdtusd.Ticker.String(): usdtusd,
+		ethusdt.Ticker.String(): ethusdt,
+	}
+)
+
+func (s *KeeperTestSuite) TestGets() {
+	s.Run("get no tickers", func() {
+		got, err := s.keeper.GetAllMarkets(s.ctx)
+		s.Require().NoError(err)
+		s.Require().Equal([]types.Market(nil), got)
+	})
+
+	s.Run("setup initial markets", func() {
+		s.Require().NoError(s.keeper.CreateMarkets(s.ctx, markets))
+
+		s.Run("unable to set markets again", func() {
+			s.Require().ErrorIs(s.keeper.CreateMarkets(s.ctx, markets), types.NewMarketAlreadyExistsError(types.TickerString(btcusdt.Ticker.String())))
+		})
+	})
+
+	s.Run("get all tickers", func() {
+		got, err := s.keeper.GetAllMarketsMap(s.ctx)
+		s.Require().NoError(err)
+
+		s.Require().Equal(len(markets), len(got))
+		s.Require().Equal(marketsMap, got)
+	})
+}
```

Comment:
Can you add test coverage for:
1. Market that has a normalize by pair that does not exist in state.
2. Updating a single market.
3. Updating a market that then includes a normalize by pair that does not exist.

Reply:
https://github.com/skip-mev/slinky/pull/270/commits/f10c268149233c5ed6b76981f5a9f91d3148804a